pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -89,10 +92,13 @@ func (r *ReconcileExtendedStatefulSet) Reconcile(request reconcile.Request) (rec
 		// If it doesn't exist, create it
 		r.log.Info("StatefulSet '", desiredStatefulSet.Name, "' for ExtendedStatefulSet '", request.NamespacedName, "' not found, will be created.")
 
+		// Reserve user-defined template
+		reservedTemplate := exStatefulSet.Spec.Template.DeepCopy()
```

Comment:
Why is this needed?

Reply:
Hi, @aduffeck. I found that exStatefulSet will obtain three default values: `ImagePullPolicy `, `TerminationMessagePath ` and `TerminationMessagePolicy ` if `spec.template` doesn't speficy after creating its statefulSet: https://github.com/kubernetes/kubernetes/blob/77a1199b6cba0df666d0a830483e4c725fe3b444/pkg/apis/core/v1/defaults.go#L77-L95

Then controller will update exStatefulSet with inconsistent template sha1sum. When reconcile loop continue, reconcile would create next version because `spec.template` has been changed.

So I just keep `spec.template`  which I mean is user-defined.